src/akkudoktoreos/class_load_container.py

```diff
@@ -1,39 +1,47 @@
 
 
-class Gesamtlast:
-    def __init__(self, prediction_hours=24):
-        self.lasten = {}  # Contains names and load arrays for different sources
-        self.prediction_hours = prediction_hours
+class LoadAggregator:
+    def __init__(self, prediction_hours: int = 24) -> None:
+        """
+        Initializes the LoadAggregator object with the number of prediction hours.
+
+        :param prediction_hours: Number of hours to predict (default: 24)
+        """
+        self.loads: Dict[
+            str, List[float]
+        ] = {}  # Dictionary to hold load arrays for different sources
+        self.prediction_hours: int = prediction_hours
 
-    def hinzufuegen(self, name, last_array):
+    def add_load(self, name: str, last_array: Union[List[float], Tuple[float, ...]]) -> None:
         """
-        Adds an array of loads for a specific source.
+        Adds a load array for a specific source. Accepts either a Python list or tuple.
 
-        :param name: Name of the load source (e.g., "Household", "Heat Pump")
-        :param last_array: Array of loads, where each entry corresponds to an hour
+        :param name: Name of the load source (e.g., "Household", "Heat Pump").
+        :param last_array: List or tuple of loads, where each entry corresponds to an hour.
+        :raises ValueError: If the length of last_array doesn't match the prediction hours.
         """
+        # Check length of the array without converting
         if len(last_array) != self.prediction_hours:
             raise ValueError(f"Total load inconsistent lengths in arrays: {name} {len(last_array)}")
-        self.lasten[name] = last_array
+        self.loads[name] = list(last_array)
 
-    def gesamtlast_berechnen(self):
+    def calculate_total_load(self) -> List[float]:
         """
-        Calculates the total load for each hour and returns an array of total loads.
+        Calculates the total load for each hour by summing up the loads from all sources.
 
-        :return: Array of total loads, where each entry corresponds to an hour
+        :return: A list representing the total load for each hour.
+                 Returns an empty list if no loads have been added.
         """
-        if not self.lasten:
-            return []
+        if not self.loads:
+            return []  # Return empty list if no loads are present
 
-        # Assumption: All load arrays have the same length
-        stunden = len(next(iter(self.lasten.values())))
-        gesamtlast_array = [0] * stunden
+        # Initialize the total load array with zeros
+        LoadAggregator_array = [0.0] * self.prediction_hours
 
-        for last_array in self.lasten.values():
-            gesamtlast_array = [
-                gesamtlast + stundenlast
-                for gesamtlast, stundenlast in zip(gesamtlast_array, last_array)
-            ]
+        # Sum loads from all sources
+        for last_array in self.loads.values():
+            for i in range(self.prediction_hours):
```

Comment:
I'm not sure if this is sensfull. The Load Container is now dependend on prediction_hours. I think it's not until now.

Reply:
removed prediction_hours hers. also used numpy again to make it fast.